tests/subscribe.spec.ts

```diff
@@ -26,10 +26,117 @@ test('it should subscribe and unsubscribe', async ({
     // Clipboard doesn't work here. Remove this when we moved away from Clipboard in favor of links
     test.skip()
   }
-  await modalPage.promptSiwe()
+  await inboxPage.promptSiwe()
   await walletValidator.expectReceivedSign({})
   await walletPage.handleRequest({ accept: true })
-  await modalPage.rejectNotifications()
-  await modalPage.subscribe(0)
-  await modalPage.unsubscribe(0)
+  await inboxPage.rejectNotifications()
+  await inboxPage.subscribeAndNavigateToDapp(0)
+  await inboxPage.unsubscribe()
+})
+
+test('it should subscribe, update preferences and unsubscribe', async ({
+  inboxPage,
+  walletPage,
+  walletValidator,
+  browserName
+}) => {
+  if (browserName === 'webkit') {
+    // Clipboard doesn't work here. Remove this when we moved away from Clipboard in favor of links
+    test.skip()
+  }
+  await inboxPage.promptSiwe()
+  await walletValidator.expectReceivedSign({})
+  await walletPage.handleRequest({ accept: true })
+  await inboxPage.rejectNotifications()
+  await inboxPage.subscribeAndNavigateToDapp(0)
+  await inboxPage.updatePreferences()
+  await inboxPage.unsubscribe()
+})
+
+test('it should subscribe and unsubscribe to and from multiple dapps', async ({
+  inboxPage,
+  walletPage,
+  walletValidator,
+  browserName
+}) => {
+  if (browserName === 'webkit') {
+    // Clipboard doesn't work here. Remove this when we moved away from Clipboard in favor of links
+    test.skip()
+  }
+  await inboxPage.promptSiwe()
+  await walletValidator.expectReceivedSign({})
+  await walletPage.handleRequest({ accept: true })
+  await inboxPage.rejectNotifications()
+  await inboxPage.subscribe(0)
+  await inboxPage.subscribe(1)
+
+  // Wait for the 2 dapps to be subscribed to.
+  await inboxPage.page.waitForFunction(() => {
```

Comment:
Maybe we could turn this into a function like `waitForSubscription` and it could be called like the following, wdyt?

```js
await inboxPage.waitForSubscriptions(2); 
```

The function could be something like this:

```js
/**
 * Waits for a specific number of dApps to be subscribed.
 * 
 * @param {object} page - The page object.
 * @param {number} expectedCount - The expected number of dApps to wait for.
 * @returns {Promise<void>}
 */
async function waitForSubscriptions(page, expectedCount) {
  // Wait for a function that checks the length of a list or a set of elements
  // matching a certain condition to equal the expectedCount.
  await page.waitForFunction(([selector, count]) => {
    const elements = document.querySelectorAll(selector);
    return elements.length === count;
  }, ['.AppSelector__list:nth-of-type(2)', expectedCount], { timeout: 5000 }); // Adjust the selector and timeout as necessary.
}
```